src/command_ext.rs

```diff
@@ -35,42 +27,3 @@ impl CommandExt for StdCommand {
         shell_words::join(tokens)
     }
 }
-
-pub trait SpawnExt {
-    /// The type of spawned processes.
-    type Child;
-
-    /// Spawn the command, but do not inherit `SIGINT` signals from the calling process.
-    fn spawn_group_without_inheriting_sigint(&mut self) -> miette::Result<Self::Child>;
-}
-
-impl SpawnExt for Command {
-    type Child = AsyncGroupChild;
-
-    fn spawn_group_without_inheriting_sigint(&mut self) -> miette::Result<Self::Child> {
-        spawn_without_inheriting_sigint(|| {
-            self.group_spawn()
-                .into_diagnostic()
-                .wrap_err_with(|| format!("Failed to start `{}`", self.display()))
-        })
-    }
-}
-
-fn spawn_without_inheriting_sigint<T>(
-    spawn: impl FnOnce() -> miette::Result<T>,
-) -> miette::Result<T> {
-    // See: https://github.com/rust-lang/rust/pull/100737#issuecomment-1445257548
-    let mut old_signal_mask = SigSet::empty();
-    pthread_sigmask(
```

Comment:
It turns out that this doesn't just stop `ghci` from _inheriting_ `SIGINT` from `ghciwatch`, it also stops `ghci` from receiving `SIGINT` whatsoever.

This is a little messier, I guess — when users press Ctrl-C the underlying `ghci` session will see that as well — but then `ghciwatch` will shut down, so it doesn't matter much in the end.